internal/server/server.go

```diff
@@ -32,3 +41,76 @@ func New(logger *log.Logger, client Client, templates map[string]*template.Templ
 
 	return http.StripPrefix(prefix, mux)
 }
+
+type RedirectError string
+
+func (e RedirectError) Error() string {
+	return "redirect to " + string(e)
+}
+
+func (e RedirectError) To() string {
+	return string(e)
+}
+
+type StatusError int
+
+func (e StatusError) Error() string {
+	code := e.Code()
+
+	return fmt.Sprintf("%d %s", code, http.StatusText(code))
+}
+
+func (e StatusError) Code() int {
+	return int(e)
+}
+
+type Handler func(w http.ResponseWriter, r *http.Request) error
+
+func errorHandler(name string, logger *log.Logger, tmplError Template, prefix, siriusURL string, next Handler) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		if err := next(w, r); err != nil {
+			type v struct {
+				SiriusURL string
+				Path      string
+				Code      int
+				Error     string
+			}
+
+			if err == sirius.ErrUnauthorized {
+				http.Redirect(w, r, siriusURL+"/auth", http.StatusFound)
+				return
+			}
+
+			if redirect, ok := err.(RedirectError); ok {
+				http.Redirect(w, r, prefix+redirect.To(), http.StatusFound)
+				return
+			}
+
+			code := http.StatusInternalServerError
+
+			if status, ok := err.(StatusError); ok {
+				if status.Code() == http.StatusForbidden || status.Code() == http.StatusNotFound {
```

Comment:
Could we just print the generic template for all other status codes?

Reply:
My logic was that something like a 400 should already be handled, but I guess that tallies: if we get to this stage and the only information we have is that a 400 occurred, then that's pretty much a server error.